### ISSUE TITLE
Nazione default value non valorizzato nella serialize e deserialize xml

### ISSUE THREAD

**Issue by pasqualecavone** [FatturaElettronica/FatturaElettronica.NET#413]
Nello schema xsd Schema_VFPR12.xsd il tag Nazione è obbligatorio ma con default IT nel type IndirizzoType.
Dallo sdi possono arrivare fatture con CedentePrestatore\Sede valorizzati con tag Nazione vuoto e stessa cosa può essere fatta in invio a sdi.

Esempio
<Sede>
  <Indirizzo>VIALE ROMA 543</Indirizzo>
  <CAP>07100</CAP>
  <Comune>SASSARI</Comune>
  <Provincia>SS</Provincia>
  <Nazione/>
</Sede>

Mi aspetto che deserializzando il file xml, il campo Nazione venga valorizzato con IT ma questo non avviene.
Ho provato a cambiare il file Samples/IT01234567890_FPR02.xml mettendo <Nazione/> e il test method DeserializeAndThenSerializeOfficialPRSample (class XMLDeSerialize) fallisce.
Stessa logica nella serialize, ovvero se l'oggetto IndirizzoType è presente allora se nazione non valorizzata deve avere la stringa di default.
Grazie

**Comment by michaelmairegger** [comment]
Sebbene non sia lo sviluppatore di questa libreria, penso che la serializzazione di A(xml) -> B -> C (xml) dovrebbe essere fatta in modo tale che A == C. Se la nazione è impostata su IT durante la deserializzazione da A a B, dopo la serializzazione da B a C l'XML risultante non sarà più identico ad A

**Comment by pasqualecavone** [comment]
Concordo con A == C, ma quindi dovrei aspettarmi che se A(xml) ha `<Nazione/>` allora anche C (xml) deve avere `<Nazione/>` e questo non avviene.
In aggiunta, si potrebbe pensare di sviluppare i metodi ReadXmlValueOrDefault e WriteXmlValueOrDefault in modo che 
A(xml con `<Nazione/>`) -> B (classe con Nazione="IT") -> C (xml con `<Nazione>IT</Nazione>`) -> D (classe con Nazione="IT")
si ha B == D

**Comment by nicolaiarocci** [comment]
Ciao grazie @pasqualecavone per la segnalazione. 

- In deserializzazione di `<Nazione />` dovremmo probabilmente rendere il valore "IT", come da specifiche xsd, anche perchè validazione fallisce erroneamente (validazione viene fatta su istanza);
- In serializzazione non possiamo far altro che continuare a fare quel che già facciamo, ovvero serializzare il valore di Nazione come è. Ne risulterebbe che A(xml) -> istanza -> B(xml) crea B != A, ma semanticamente i due XML hanno contenuti (significati) identici.

Insomma, mi sembra più importante aderire alle specifiche e convalidare un XML con tag Nazione vuoto, anche perchè le conseguenze negative sono minime.

**Comment by pasqualecavone** [comment]
Grazie, sono concorde con la tua analisi.
